sql/event_data_objects.cc

```diff
@@ -194,6 +194,7 @@ Event_queue_element_for_exec::init(const LEX_CSTRING &db, const LEX_CSTRING &n)
                              n.str, name.length= n.length, MYF(MY_WME))))
   {
     my_free(const_cast<char*>(dbname.str));
+    dbname_str_freed= true;
```

Comment:
I think you can just `dbname.str= nullptr` because `my_free()` supports  `nullptr` arguments. This way you won't need a `dbname_str_freed` variable.

Reply:
https://github.com/MariaDB/server/commit/6244876488f41c4aec8e55ede37625b5bc649bbd

Does that, also removes the potential memory leak if init fails.

Reply:
Nice! BTW, in case of allocation failure `new` will `throw std::bad_alloc` and server will die from unhandled exception.